### PR TITLE
No sandbox memory instructions

### PR DESCRIPTION
**Pull request by matthias-blume**
This implements no-sandboxing for memory loads and stores.

If these instructions load from locations that correspond to global data, they are rewritten to use the data in generated data segments.

This PR comes logically after no-sandbox-const-instructions.
It also relies on the yet-to-be-implemented change to how data segments are treated in no-sandbox mode.

### CODE REVIEW

File: src/c-writer.cc [hunk 1]
```diff
@@ -3380,6 +3388,92 @@ void CWriter::Write(const ConvertExpr& expr) {
...
+  return true;
+}
+
+void CWriter::WriteMemoryAddress(Index stack_index,
+                                 const Memory* memory,
+                                 size_t loc_offset,
+                                 Address offset) {
+  if (options_.no_sandbox) {
+    Write("(u64)(", StackVar(stack_index), ")");
+    if (!MaybeWriteNoSandboxMemoryAddress(loc_offset, memory->page_limits.is_64,
```
Comment (dimitry-) [inline src/c-writer.cc]: `if (Maybe ...) return;`

File: src/c-writer.cc [hunk 2]
```diff
@@ -3380,6 +3388,92 @@ void CWriter::Write(const ConvertExpr& expr) {
...
+      if (offset != 0) {
+        Write(" + ", offset, "u");
+      }
+    }
+  } else {
+    Write(ExternalInstancePtr(ModuleFieldType::Memory, memory->name),
+          ", (u64)(", StackVar(stack_index), ")");
+    if (offset != 0) {
+      Write(" + ", offset, "u");
+    }
```
Comment (dimitry-) [inline src/c-writer.cc]: and move this out of if.

**Review by dimitry- — COMMENTED** [review]
otherwise lgtm.

File: src/c-writer.cc (hunk 1 above)
Comment (matthias-blume) [reply src/c-writer.cc]: done

File: src/c-writer.cc (hunk 2 above)
Comment (matthias-blume) [reply src/c-writer.cc]: done

**Comment by matthias-blume** [comment]
ptal

**Review by dimitry- — APPROVED** [review]

**Pull request merged**